mypy.ini

```diff
@@ -1,5 +1,5 @@
 [mypy]
-python_version = 3.9
+python_version = 3.8
```

Comment:
Python 3.8 does not support subscripting builtin types like `list` and `dict`. So you have 3 options:
1. `from __future__ import annotations` at the top of each file, this stops Python from evaluating types at runtime
2. escape types in strings, e.g. `def foo(a: "list[str]") -> "dict[str, str]": ...`
3. Use `typing` instead, e.g. `list` -> `typing.List`

I personally prefer the future annotations import, as it generally makes working with types a bit cleaner overall, but up to you!

Reply:
Thanks very much! In this PR, sticking to using `typing` and `typing.List` / `typing.Dict` as that's what we've been using thus far, but maybe will switch to 1. in future.